pom.xml

```diff
@@ -94,6 +94,7 @@
         <org.slf4j.version>1.7.25</org.slf4j.version>
         <ranger.version>0.7.1</ranger.version>
         <jetty.version>9.4.3.v20170317</jetty.version>
+        <httpclient.version>4.5.5</httpclient.version>
```

Comment:
Per [NIFI-4936](https://issues.apache.org/jira/browse/NIFI-4936), this needs to be pushed down to each NAR/bundle. In your case since you are using it in nifi-hadoop-utils, each NAR that has this as a dependency should declare it. If they already bring in httpclient, you'd likely need to reuse that version rather than setting it to 4.5.5, to avoid eviction.

Reply:
I'm not sure I got this issue right. I removed this property and replaced it to a direct version specification where it was used. Is that enough?

Reply:
I think I see the issue now. I changed the httpclient and hadoop-auth dependency to "provided" in nifi-hadoop-util so they won't be included into all NARs depending on nifi-hadoop-util.

Reply:
@mattyb149, could you please review my change and let me know if I did the right thing? Any suggestions are welcome.

I changed the httpclient dependency to provided so it won't get into all NARs depending on nifi-hadoop-utils. Unfortunately httpclient seems to change a lot from version to version and the minimum required version that my changes can work with is 4.4.1.

All my additions to hadoop-utils are in new classes, so I believe even a hadoop-utils dependent NAR that specifies a lower version of httpclient will be fine as long as it doesn't start to use my classes directly.